yetanotherco/aligned_layer
# fix: consolidate 2-step listen after sending proofs

# Consolidate 2-step listen after sending proofs

## Description

This PR fixes a bug found when stress testing the batcher. The batcher would respond with `BatchInclusionData` to a websocket that was still sending proofs, expecting more `ValidityResponseMessage`s . This case errored with an undescriptive `Serialization error`

Update: The fix to this issue will be to remove the 2-step listening from the sender side, to only listen 1 message type, either proof rejected by x reason or proof is valid and included in x merkle_root.

These 2 steps are sequencial (first send the burst of proofs, then read their responses).

This PR also refactors some of the nearby code, hopefully making it more understandable.

## Type of change


## Checklist
- [x] Linked to Github Issue #1314 

## To Test

- Make sure everything works normally
- it is important to send different proofs in different ways, because this refactors how the senders sends the proofs
- for example, sending different bursts
  - `batcher_send_sp1_burst`
  - `batcher_send_risc0_burst`
  - `batcher_send_plonk_bn254_burst`
  - `batcher_send_plonk_bls12_381_burst`
  - `batcher_send_burst_groth16`
- sending by individual tasks
  - `batcher_send_sp1_task`
  - `batcher_send_risc0_task` or `batcher_send_risc0_task_no_pub_input`
  - `batcher_send_plonk_bn254_task`
  - `batcher_send_plonk_bls12_381_task`
  - `batcher_send_groth16_bn254_task`
- sending infinite
  - `batcher_send_infinite_sp1`
  - `batcher_send_infinite_groth16`

## To Deploy
- SDK
  - Batcher
  - CLI (minor error format change from SDK)
